### PR TITLE
decompress, treating None and 0 differently

### PR DESCRIPTION
**Pull request by mikeford3**
When getting a value from a database 0 is treated as null and therefore not loaded/shown in a form.

Fix suggested to check for null explicitly

### CODE REVIEW

**Comment by CarliJoy** [comment]
Thanks for figuring out this bug (yet again one).
Due to help of contributors like you, OS stuff gets better :-)

I will have a look at the PR soon.

**Comment by CarliJoy** [comment]
On thing I am confused about why are you using two different accounts? mikeford1 and mikeford3 ?

File: src/quantityfield/widgets.py [hunk 1]
```diff
@@ -24,7 +24,7 @@ def decompress(self, value):
 
         It is responsible to split values for the two widgets
         """
-        if value:
+        if value is not None:
```
Comment (CarliJoy) [inline src/quantityfield/widgets.py:27]:
```suggestion
    # TODO Please add `import numbers` on the top
    # We check if the input is a number and continue, may be we have to consider strings that can
    # be converted to floats too (I hope not)
    # Details see: https://stackoverflow.com/questions/4187185/how-can-i-check-if-my-python-object-is-a-number
        if isinstance(value, numbers.Number):
```

File: tests/test_widget.py [hunk 2]
```diff
@@ -282,3 +282,12 @@ class TestWidgetRenderingSmallNumber(TestWidgetRenderingBase):
...
     expected_db = "1e-10"
+
+
+class TestWidgetRenderingZero(TestWidgetRenderingBase):
+    value = 0
+    expected_created = "0"
+    expected_db = "0.0"
+
+
+# not sure what a db should return if the value is None or ""?
```
Comment (CarliJoy) [inline tests/test_widget.py:293]:
```suggestion
```
Should not matter if we test for numbers, but good hint.

Please also create a test for `0.0` and `Decimal("0.0"). Thank you :-)

Good Work

**Review by CarliJoy — CHANGES_REQUESTED** [review]
I would generalize the code a bit, as you found these issues arleady.

**Comment by mikeford3** [comment]
Thanks for the very prompt review, I'll update as suggested.

Not sure why I'm showing as mike.ford1 & mike.ford3, probably something to do with my local git settings - I'll fix it and then resubmit a different PR.

**Pull request closed without merging**